packages/nx/src/project-graph/utils/project-configuration-utils.ts

```diff
@@ -434,15 +451,18 @@ export async function createProjectConfigurations(
       Object.assign(externalNodes, pluginExternalNodes);
     }
 
-    let projects: Record<string, ProjectConfiguration>;
     try {
-      projects = readProjectConfigurationsFromRootMap(projectRootMap);
+      // We still call this just to assert that the root map
+      // only contains valid project names. This is a safety check.
+      //
+      // The signature itself can't be changed as we need it to return
+      // project configurations for use in devkit.
+      readProjectConfigurationsFromRootMap(projectRootMap);
```

Comment:
Can you change the name of the function to something that begins with validate?

Reply:
We still use the function in 2-3 other places to read the configs, so I don't think we'd change it